### PR TITLE
feat(ratelimit): traffic limit http api

### PR DESCRIPTION
**Pull request by hunshenshi**
fork from #649  and #648 
issues #647 
add http api for traffic limit

### CODE REVIEW

**Comment by Liuhaai** [comment]
IMO, rate limiting for api would be better to be done at Cloudflare or nginx. Otherwise, how could the requests be limited per IP in the backend?

**Comment by hunshenshi** [comment]
> IMO, rate limiting for api would be better to be done at Cloudflare or nginx. Otherwise, how could the requests be limited per IP in the backend?

It is possible to set different rates for each project, similar to a billing system.

File: cmd/srv-applet-mgr/apis/middleware/current_user.go [hunk 1]
```diff
@@ -228,3 +229,24 @@ func (v *CurrentAccount) WithChainTxBySFID(ctx context.Context, id types.SFID) (
...
+}
+
+func (v *CurrentAccount) WithTrafficLimitContextBySFIDAndProjectName(ctx context.Context, id types.SFID) (context.Context, error) {
+	traffic, err := trafficlimit.GetBySFID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	project := types.MustProjectFromContext(ctx)
+	if traffic.ProjectID != project.ProjectID {
+		return nil, status.NoTrafficLimitPermission
```
Comment (saitofun) [inline cmd/srv-applet-mgr/apis/middleware/current_user.go:249]: NoProjectPermission

**Review by saitofun — CHANGES_REQUESTED** [review]
gen oas manual

**Review by saitofun — APPROVED** [review]

**Pull request merged**